### PR TITLE
Implement Basic Info Module

### PR DESCRIPTION
**Pull request by NathHorrigan**
This PR add's the basic info module which records info such as `dob`, `headphoneType`, `operatingSystem`.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -27,9 +29,12 @@
     "react-dom": "~16.13.0",
     "react-native": "~0.63.3",
     "react-native-config": "^1.4.0",
+    "react-native-date-picker": "^3.2.5",
     "react-native-dotenv": "^2.4.1",
     "react-native-gesture-handler": "^1.7.0",
     "react-native-markdown-display": "^7.0.0-alpha.2",
+    "react-native-modalize": "^2.0.8",
+    "react-native-picker-select": "^8.0.4",
```
Comment (mixxorz) [inline package.json:36]: Just double checking if we need all of these pickers.

File: src/components/TextField.tsx [hunk 2]
```diff
@@ -32,3 +36,86 @@ export const TextField = createRestyleComponent<Props, Theme>(
...
+          height: 50,
+          borderRadius: 8,
+          paddingLeft: 16,
+        }}
+        onValueChange={onChange}
+        items={options}
+      />
+    </Box>
+  </Box>
+)
```
Comment (mixxorz) [inline src/components/TextField.tsx]: Some of the components in this file feel like they don't belong inside `TextField`. Maybe generalize this file and call it `FormField` or something.

File: src/containers/BasicInfoContainer.tsx [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+      <HeadphoneChoiceScreen
+        headphoneType={mod.headphoneType}
+        updateHeadphoneType={(headphoneType) => {
+          updateModule({ headphoneType })
+          updateExperiment({ headphoneType })
+        }}
+        onNext={nextScreen}
+      />
+    )
+  }
```
Comment (mixxorz) [inline src/containers/BasicInfoContainer.tsx:78]: Shouldn't we protect against out of bounds errors here? Maybe use a switch statement?

File: src/utils/exampleExperiment.ts [hunk 4]
```diff
@@ -94,6 +94,17 @@ export const exampleExperimentData: Experiment = {
     max: 1500,
   },
   modules: [
+    {
+      id: '451651',
+      moduleType: 'BASIC_INFO',
+      definition: {
+        genders: [
+          { label: 'Male', value: 'male' },
+          { label: 'Female', value: 'male' },
```
Comment (mixxorz) [inline src/utils/exampleExperiment.ts]: `value` should be `female`

**Review by mixxorz — COMMENTED** [review]
Overall looks good. Just spotted a few things. Thanks Nathan!

File: package.json (hunk 1 above)
Comment (NathHorrigan) [reply package.json:36]: Removed `react-native-modalize`.

File: src/components/TextField.tsx (hunk 2 above)
Comment (NathHorrigan) [reply src/components/TextField.tsx]: Good Idea, Switched.

File: src/containers/BasicInfoContainer.tsx (hunk 3 above)
Comment (NathHorrigan) [reply src/containers/BasicInfoContainer.tsx:78]: Changed to switch :)

**Pull request merged**